### ISSUE TITLE
use of unititialized value message

### ISSUE THREAD

**Issue by schultzm** [lskatz/mashtree#29]
`Use of uninitialized value in concatenation (.) or string at /home/linuxbrew/.linuxbrew/Cellar/perl/5.26.2/lib/perl5/site_perl/5.26.2/Mashtree/Db.pm line 237.`

Even though I see this message, the run seems to produce a valid result.

**Comment by tseemann** [comment]
I just got it too:
```
Use of uninitialized value in concatenation (.) or string at /home/linuxbrew/.linuxbrew/Cellar/perl/5.26.2/lib/perl5/site_perl/5.26.2/Mashtree/Db.pm line 237.
```

```
  my $str=join("\t",".",@name)."\n";
  for(my $i=0;$i<$numNames;$i++){
    $str.=$name[$i];
    for(my $j=0;$j<$numNames;$j++){
      $str.="\t".$distance{$name[$i]}{$name[$j]};
    }
    $str.="\n";
  }
```

**Comment by lskatz** [comment]
Is this with the latest version?  And do you think it is data-dependent?

**Comment by lskatz** [comment]
Actually.... this was a sneaky bug that might cause issues when you are specifying `--outmatrix`. I just figured it out.  Below is an example of the underlying perl hash.  In about half the cases, a value would not be found because it was not explicitly found.  For example, the distance between sample 3 and sample 1 was found, but not the distance between sample 1 and sample 3.  I am fixing this in e83de4babf6abd9f2eee840b634a0b653eb61bef.

```
$VAR1 = {
          'sample4' => {
                         'sample4' => 0,
                         'sample2' => '0.00417555',
                         'sample1' => '0.0042153'
                       },
          'sample2' => {
                         'sample2' => 0,
                         'sample1' => '0.00414809'
                       },
          'sample3' => {
                         'sample4' => '0.00402957',
                         'sample2' => '0.00405078',
                         'sample3' => 0,
                         'sample1' => '0.0041298'
                       },
          'sample1' => {
                         'sample1' => 0
                       }
        };
```

**Comment by lskatz** [comment]
Fixed in version 0.33 with a unit test to make sure the matrix is being generated correctly.  A CPAN version is being rolled out...
